File: mobile/lib/features/register/register_model.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class RegisterModel {
  static Future registerEmail(String email, String password) async {
    final response = await http.post(
      Uri.parse('https://nodejsdeployowl.et.r.appspot.com/register'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(
        {
          'email': email,
          'password': password,
        },
      ),
    );

    final responseBody = jsonDecode(response.body);
    return responseBody;
  }
}

File: web/src/Features/Register/RegisterModel.js
// const registerEmail = async (email, password) => {
//     let response = await fetch('https://nodejsdeployowl.et.r.appspot.com/register', {
//         method: 'POST',
//         headers: {"Content-Type": "application/json"},
//         body: JSON.stringify({
//             'email': email,
//             'password': password
//         }),
//         redirect: 'follow'
//     }).then((response) =>
//         console.log(response.text())
//     ).then(()=>{
//         window.location.href = '/login'
//     }).catch((err) => console.log(err))
// }

const registerEmail = async (email, password) => {
    let response = await fetch('https://nodejsdeployowl.et.r.appspot.com/register', {
        method: 'POST',
        headers: {"Content-Type": "application/json"},
        body: JSON.stringify({
            'email': email,
            'password': password
        }),
        redirect: 'follow'
    })

    let responseBody = await response.json()
    console.log(responseBody)
    return responseBody
}

export {registerEmail}